lib/query/token-visitors.js

```diff
@@ -0,0 +1,188 @@
+const { isEmpty, compact, isNil, isFunction, isString, cloneDeep, castArray } = require("lodash");
+const Tokenizer = require("tokenize-this");
+const { operators } = require("../operators");
+
+class TokenListVisitor {
+    constructor(tokens) {
+        if (isString(tokens)) tokens = this.tokenize(tokens);
+        this.tokens = tokens;
+    }
+    tokenize(rawStr) {
+        const tokens = [];
+        const tokenizer = new Tokenizer();
+        tokenizer.tokenize(rawStr, (token) => {
+            if (tokens.length > 0) {
+                for (const combined of [
+                    tokens[tokens.length - 1] + token,
+                    tokens[tokens.length - 1] + ' ' + token
+                ]) {
+                    if (operators[combined]) {
+                        tokens[tokens.length - 1] = combined;
+                        return;
+                    }
+                }
+            }
+            tokens.push(`${token}`);
```

Comment:
why wrap in template?

Reply:
It was done to stringify. Its no longer necessary after some other changes. removed.